### ISSUE TITLE
Fix image rotation bug

### ISSUE THREAD

**Issue by fakhirsh** [fakhirsh/cyclicshift#11]
I need to rotate the images 90 degrees clock wise or counter clock wise, as follows:

![bitmap](https://user-images.githubusercontent.com/3814383/68524915-1a018280-02ee-11ea-9706-e53d40a65c1d.png)

Some images get totally messed up when rotated 90 degrees clockwise (or counter clockwise):

**_Sample 1:_**
![img2](https://user-images.githubusercontent.com/3814383/68524940-5e8d1e00-02ee-11ea-9a0a-fb3b197ab5d4.png)  
**_Sample 2 from a different portion of the image:_**
![img4](https://user-images.githubusercontent.com/3814383/68525060-70bb8c00-02ef-11ea-83c7-8b0ef339a42f.png)

Here is the code. Find bug and correct it:

![img3](https://user-images.githubusercontent.com/3814383/68524964-a01dc900-02ee-11ea-9cb4-47283d4e8d41.png)

**Comment by mohammadusman666** [comment]
I tried to mimic what you were doing in Python and then started to correct the clockwise rotation function. The attached zip file contains fs.py file in which I commented about the changes I made. I could have done it in C++ but converting my code to C++ wouldn't take more than 5 mins at max.

[ClockwiseRotation.zip](https://github.com/fakhirsh/cyclicshift/files/3827496/ClockwiseRotation.zip)
